### ISSUE TITLE
Codefix: convert `Type[]` to `readonly Type[]`

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [microsoft/TypeScript#31923]
## Search Terms

codefix readonlyarray readonly array immutable

## Suggestion

It'd be nice to have a codefix for converting non-readonly arrays to readonly. 

## Use Cases

When converting/refactoring code to be more immutable/`readonly`-friendly, it's a bit of a pain to manually add these annotations.

## Examples

Given this:

```ts
function takesArrays(items: /*| |*/string[], moreItems: /*| |*/string[]) { /* ... */ }
```

...having a codefix to make one or all parameters `readonly` would be useful:

```ts
function takesArrays(items: readonly string[], moreItems: readonly string[]) { /* ... */ }
```

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by jcalz** [comment]
I agree, especially with the introduction of `const` assertions (#29510) and their `readonly` tuples.
